modules/python/python-persist.c

```diff
@@ -21,8 +21,49 @@
  */
 
 #include "python-persist.h"
+#include "persistable-state-header.h"
 #include "python-helpers.h"
+#include "syslog-ng.h"
 #include "driver.h"
+#include "mainloop.h"
+#include "compat/compat-python.h"
+
+#include <structmember.h>
+
+#define SUBKEY_DELIMITER "##"
+#define MASTER_ENTRY_VERSION 1
+
+typedef PersistableStateHeader PythonPersistMasterEntry;
+
+typedef enum
+{
+  ENTRY_TYPE_STRING,
+  ENTRY_TYPE_LONG,
+  ENTRY_TYPE_BYTES,
+  ENTRY_TYPE_MAX
+} EntryType;
+
+typedef struct
+{
+  guint8 type;
+  gchar data[];
```

Comment:
I think this may not be portable to older c compilers. doing it as "gchar data[0]" on the other hand is portable and shouldn't be more difficult implementation wise.

Reply:
Thanks, fixed.